src/communication_managers/iothub_communication_manager/src/iothub_communication_manager.c

```diff
@@ -725,9 +726,14 @@ static void Connection_Maintenance()
             additionalDelayInSeconds = TIME_SPAN_FIVE_MINUTES_IN_SECONDS;
             break;
         case IOTHUB_CLIENT_CONNECTION_NO_NETWORK:
-            // Could be transient error, wait for at least 5 minutes to retry.
+            // 1) try to restart agent to prevent empty reported properties in module twin 
+            // 2) if restart agent could not be performed, wait for at least 5 minutes to retry.
             Log_Error("No network.");
-            additionalDelayInSeconds = TIME_SPAN_FIVE_MINUTES_IN_SECONDS;
+            if (ADUC_MethodCall_RestartAgent() != 0)
```

Comment:
NoNetwork has a lot of meanings in the iot-hub-c-sdk. I'm ok with a hard restart as an option, but do we have a max retries for restarting the daemon after a failure? I know we can rely on the exponential back off of the daemon for the retry stuff but what happens if we hit our (5 I believe) max retries? Do we have a restart story?

Reply:
the default value of the deviceupdate-agent.service is to restart always after a timeout of 5s. https://github.com/JoergZeidler/iot-hub-device-update-azure/blob/e0c7f3b83292479c0f4666fbaf3e53d8303c7f66/daemon/deviceupdate-agent.service#L8